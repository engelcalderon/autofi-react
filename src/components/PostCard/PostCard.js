import React from 'react';
import { object } from 'prop-types';

import css from './PostCard.module.css';

const PostCard = props => {
    const { post } = props;
    const { title, body } = post;

    return (
        <div className={css.root}>
            <div className={css.postImage} />

            <div className={css.body}>
                <h3 className={css.title}>{title}</h3>
                <div className={css.body}>{body}</div>
            </div>
        </div>
    )
};

PostCard.defaultProps = {
    post: null,
}

PostCard.propTypes = {
    post: object.isRequired,
}

export default PostCard;

