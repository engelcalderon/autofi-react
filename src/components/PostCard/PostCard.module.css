
.root {
    padding-bottom: 15px;
    margin-bottom: 15px;
    border-bottom: 1px solid black;

    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.postImage {
    flex-basis: calc(20% - 12px);
    margin-right: 12px;
    height: 150px;
    background-color: gray;
}

.body {
    flex-basis: 80%;
}

.title {
    margin: 0;
    padding: 0 0 15px 0;
}

.commentsContainer {
    width: 100%;
    margin-top: 10px;
    padding-left: 20%;
}

.addCommentForm {
    margin: 10px 0 15px 0;
}

.commentsLabel {
    cursor: pointer;
    color: #0000EE ;
}

.commentsLabel:hover {
    text-decoration: underline;
}