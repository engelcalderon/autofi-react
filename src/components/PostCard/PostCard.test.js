import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostCard from './PostCard';

describe('PostCard', () => {
    it('matches snapshot', () => {
        const post = {
            "userId": 1,
            "comments": [
                {
                    "postId": 1,
                    "id": 1,
                    "name": "id labore ex et quam laborum",
                    "email": "Eliseo@gardner.biz",
                    "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\ntempora quo necessitatibus\ndolor quam autem quasi\nreiciendis et nam sapiente accusantium"
                },
                {
                    "postId": 1,
                    "id": 2,
                    "name": "quo vero reiciendis velit similique earum",
                    "email": "Jayne_Kuhic@sydney.com",
                    "body": "est natus enim nihil est dolore omnis voluptatem numquam\net omnis occaecati quod ullam at\nvoluptatem error expedita pariatur\nnihil sint nostrum voluptatem reiciendis et"
                }
            ],
            "id": 1,
            "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
            "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
        };
        const tree = toJson(mount(<PostCard post={post} onAddComment={() => null} />), { mode: 'deep' });
        expect(tree).toMatchSnapshot();
    })
});