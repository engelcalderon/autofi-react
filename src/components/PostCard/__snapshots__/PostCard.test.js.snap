// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard matches snapshot 1`] = `
<div
  className="root"
>
  <div
    className="postImage"
  />
  <div
    className="body"
  >
    <h3
      className="title"
    >
      sunt aut facere repellat provident occaecati excepturi optio reprehenderit
    </h3>
    <div
      className="body"
    >
      quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
    </div>
  </div>
  <div
    className="commentsContainer"
  >
    <span
      className="commentsLabel"
      onClick={[Function]}
    >
      2 comments
    </span>
  </div>
</div>
`;
